libraries/AP_Airspeed/AP_Airspeed_SDP3X.cpp

```diff
@@ -0,0 +1,303 @@
+            _dev->get_semaphore()->give();
+            continue;
+        }
+
+        // start continuous average mode
+        if (!_send_command(SDP3X_CONT_MEAS_AVG_MODE)) {
+            _dev->get_semaphore()->give();
+            continue;
+        }
+
+        hal.scheduler->delay_microseconds(20000);
+
+        // step 3 - get scale
+        uint8_t val[9];
+        ret = _dev->transfer(nullptr, 0, &val[0], sizeof(val));
+        if (!ret) {
+            _dev->get_semaphore()->give();
+            continue;
+        }
+
+        // Check the CRC
+        if (!_crc(&val[0], 2, val[2]) || !_crc(&val[3], 2, val[5]) || !_crc(&val[6], 2, val[8])) {
+            _dev->get_semaphore()->give();
+            continue;
+        }
+
+        _scale = (((uint16_t)val[6]) << 8) | val[7];
+
+        _dev->get_semaphore()->give();
+
+        found = true;
+
+        char c = 'X';
+        switch (_scale) {
+        case SDP3X_SCALE_PRESSURE_SDP31:
+            c = '1';
+            break;
+        case SDP3X_SCALE_PRESSURE_SDP32:
+            c = '2';
+            break;
+        case SDP3X_SCALE_PRESSURE_SDP33:
+            c = '3';
+            break;
+        }
+        hal.console->printf("SDP3%c: Found on bus %u address 0x%02x scale=%u\n",
+                            c, get_bus(), addresses[i], _scale);
+    }
+
+    if (!found) {
+        return false;
+    }
+
+    /*
+      this sensor uses zero offset and skips cal
+     */
+    set_use_zero_offset();
+    set_skip_cal();
+    set_offset(0);
+    
+    // drop to 2 retries for runtime
+    _dev->set_retries(2);
+
+    _dev->register_periodic_callback(20000,
+                                     FUNCTOR_BIND_MEMBER(&AP_Airspeed_SDP3X::_timer, void));
+    return true;
+}
+
+// read the values from the sensor. Called at 50Hz
+void AP_Airspeed_SDP3X::_timer()
+{
+    // read 6 bytes from the sensor
+    uint8_t val[6];
+    int ret = _dev->transfer(nullptr, 0, &val[0], sizeof(val));
+    if (!ret) {
+        return;
+    }
+    // Check the CRC
+    if (!_crc(&val[0], 2, val[2]) || !_crc(&val[3], 2, val[5])) {
+        return;
+    }
+
+    int16_t P = (((int16_t)val[0]) << 8) | val[1];
+    int16_t temp = (((int16_t)val[3]) << 8) | val[4];
+
+    float P_float = (float)P;
+    float temp_float = (float)temp;
+
+    float scale_float = (float)_scale;
+
+    float diff_press_pa = P_float / scale_float;
+    float temperature = temp_float / (float)SDP3X_SCALE_TEMPERATURE;
+
+    if (sem->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        _press_sum += diff_press_pa;
+        _temp_sum += temperature;
+        _press_count++;
+        _temp_count++;
+        _last_sample_time_ms = AP_HAL::millis();
+        sem->give();
+    }
+}
+
+/*
+  correct pressure for barometric height
+  With thanks to:
+   https://github.com/PX4/Firmware/blob/master/Tools/models/sdp3x_pitot_model.py
+ */
+float AP_Airspeed_SDP3X::_correct_pressure(float press)
+{
+    const float tube_len = 0.2;
```

Comment:
I thought the conclusion from the dev call was to not hard code all the tube parameters?

Reply:
its an ongoing discussion with Paul
happy to add a parameter later if it turns out it is useful

Reply:
This is significantly different then the model referenced, where did the length come from?

This whole correction model appears to be only valid for the specific tube with exactly the tube lengths and diameters as supplied by the Drotek source. (It is not a model found anywhere within the SDP datasheet or application note that I can find).